walk/walk.go

```diff
@@ -279,12 +277,12 @@ func (u *UpdateFilter) shouldVisit(rel string, updateParent bool) bool {
 	}
 }
 
-func loadBuildFile(c *config.Config, pkg, dir string, ents []fs.DirEntry) (*rule.File, error) {
+func loadBuildFile(c *config.Config, pkg string, ents []fs.DirEntry) (*rule.File, error) {
 	var err error
-	readDir := dir
+	readDir := filepath.Join(c.RepoRoot, pkg)
 	readEnts := ents
 	if c.ReadBuildFilesDir != "" {
-		readDir = filepath.Join(c.ReadBuildFilesDir, filepath.FromSlash(pkg))
+		readDir = filepath.Join(c.ReadBuildFilesDir, pkg)
```

Comment:
Why is this safe? Does `pkg` contain OS-specific path separators at this point?T

Reply:
We don't do it 3 lines up (which was moved from the `loadBuildFile` call site), so I don't think we should do it here either. We could do the opposite and do it in both, but I'd rather take the one with less code if it works...

Reply:
I dropped this second commit for now.